### PR TITLE
PostTypeList: Remove feature flag and enable condensed posts globally.

### PR DESCRIPTION
**Pull request by jonathansadowski**
This PR removes the feature-flag `posts/post-type-list` and removes/refactors code that is no longer used due to its removal, and thereby enabled Condensed Post List in all environments.

Before (in environments where CPL was disabled):

image

After:

image

To test:
* Apply PR and check for regressions.

### CODE REVIEW

File: client/blocks/post-share/style.scss [hunk 1]
```diff
@@ -283,13 +263,7 @@ $section-border: solid 1px darken( $sidebar-bg-color, 5% );
...
 
 	@include breakpoint( ">480px" ) {
-		margin-right: 24px;
-		margin-left: 24px;
-
-		//TODO Clean-up after PostTypeList is used for /posts/
-		.post-type-list & {
-			margin: 6 24px 0;
-		}
+		margin: 6px 24px 0;
```
Comment (jonathansadowski) [inline client/blocks/post-share/style.scss]: @michaeldcain can you verify this change? The original looked incorrect to me so I updated it while doing the clean-up.

File: client/my-sites/posts/post.jsx [hunk 2]
```diff
@@ -1,378 +0,0 @@
...
-import { localize } from 'i18n-calypso';
-import { partial, noop } from 'lodash';
-
-/**
- * Internal dependencies
- */
-import { isEnabled } from 'config';
-import Card from 'components/card';
-import PostControls from './post-controls';
-import PostFormat from 'components/post-format';
```
Comment (nylen) [inline client/my-sites/posts/post.jsx]: This component can now be removed.

File: client/my-sites/posts/post.jsx [hunk 3]
```diff
@@ -1,378 +0,0 @@
...
-import { recordGoogleEvent, recordTracksEvent } from 'state/analytics/actions';
-import { setPreviewUrl } from 'state/ui/preview/actions';
-import { setLayoutFocus } from 'state/ui/layout-focus/actions';
-import { getPostPreviewUrl } from 'state/posts/selectors';
-import { isSingleUserSite, isSitePreviewable } from 'state/sites/selectors';
-import { getSelectedSiteId } from 'state/ui/selectors';
-import { getEditorPath } from 'state/ui/editor/selectors';
-import Comments from 'blocks/comments';
-import PostShare from 'blocks/post-share';
-import PostActions from 'blocks/post-actions';
```
Comment (nylen) [inline client/my-sites/posts/post.jsx]: This component can now be removed.

**Comment by jonathansadowski** [comment]
Nice catch @nylen, thanks! I just removed those components. I looked for more that could be removed, but nothing jumped out. Let me know if you notice anything else.

File: client/blocks/post-share/style.scss (hunk 1 above)
Comment (michaeldcain) [reply client/blocks/post-share/style.scss]: It actually looks like you can remove the entire `.post-share__services-add` selector - it doesn't seem to be used anywhere in the code (it was replaced with `.post-share__services-add-button`, which was even removed in #20912).

**Comment by jonathansadowski** [comment]
Thanks for the review.

**Pull request merged**